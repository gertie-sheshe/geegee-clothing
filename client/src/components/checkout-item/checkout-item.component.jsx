import React from 'react';
import { connect } from 'react-redux';

import {
  removeItemFromCart,
  addItemToCart,
  removeItemQuantity,
} from '../../redux/cart/cart.actions';

import {
  CheckoutItemContainer,
  ImageContainer,
  TextContainer,
  QuantityContainer,
  RemoveButtonContainer,
} from './checkout-item.styles';

const CheckOutItem = ({ cartItem, removeItem, addItem, removeQuantity }) => {
  const { name, imageUrl, price, quantity } = cartItem;
  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <img src={imageUrl} alt="item" />
      </ImageContainer>
      <TextContainer>{name}</TextContainer>
      <QuantityContainer>
        <div onClick={() => removeQuantity(cartItem)} className="arrow">
          &#10094;
        </div>
        <span className="value">{quantity}</span>
        <div onClick={() => addItem(cartItem)} className="arrow">
          &#10095;
        </div>
      </QuantityContainer>
      <TextContainer>{price * quantity}</TextContainer>
      <RemoveButtonContainer onClick={() => removeItem(cartItem)}>
        &#10005;
      </RemoveButtonContainer>
    </CheckoutItemContainer>
  );
};

const mapDispatchToProps = dispatch => ({
  removeItem: item => dispatch(removeItemFromCart(item)),
  addItem: item => dispatch(addItemToCart(item)),
  removeQuantity: item => dispatch(removeItemQuantity(item)),
});

export default connect(null, mapDispatchToProps)(CheckOutItem);
