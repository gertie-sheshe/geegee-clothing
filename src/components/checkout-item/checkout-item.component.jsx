import React from 'react';
import { connect}  from 'react-redux';

import { removeItemFromCart } from '../../redux/cart/cart.actions'

import './checkout-item.styles.scss';

const CheckOutItem = ({cartItem, removeItem}) => {
    const {name, imageUrl, price, quantity} = cartItem;
    return (
  <div className="checkout-item">
    <div className="image-container">
      <img src={imageUrl} alt="item" />
    </div>
    <span className="name">{name}</span>
    <span className="quantity">{quantity}</span>
    <span className="price">{price}</span>
    <div className="remove-button" onClick={() => removeItem(cartItem)}>&#10005;</div>
  </div>
)};

const mapDispatchToProps = dispatch => ({
    removeItem: item => dispatch(removeItemFromCart(item))
});

export default connect(null, mapDispatchToProps)(CheckOutItem);
