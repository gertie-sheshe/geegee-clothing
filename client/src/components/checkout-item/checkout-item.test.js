import React from 'react';
import { shallow } from 'enzyme';

import { CheckOutItem } from './checkout-item.component';

describe('CheckoutItem Component', () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;
  let mockRemoveQuantity;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();
    mockRemoveQuantity = jest.fn();

    const mockProps = {
      cartItem: {
        imageUrl: 'www.testimage.com',
        price: 10,
        name: 'hats',
        quantity: 2,
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
      removeQuantity: mockRemoveQuantity,
    };

    wrapper = shallow(<CheckOutItem {...mockProps} />);
  });

  it('Should render CheckoutItem Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call clearItem when remove button is clicked', () => {
    wrapper.find('RemoveButtonContainer').simulate('click');
    expect(mockRemoveItem).toHaveBeenCalled();
  });

  it('Should call removeItem when left arrow is clicked', () => {
    wrapper
      .find('QuantityContainer')
      .childAt(0)
      .simulate('click');

    expect(mockRemoveQuantity).toHaveBeenCalled();
  });

  it('Shoulc call addItem when right arrow is clicked', () => {
    wrapper
      .find('QuantityContainer')
      .childAt(2)
      .simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });
});
