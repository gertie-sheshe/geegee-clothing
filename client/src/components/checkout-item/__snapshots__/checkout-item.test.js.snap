// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem Component Should render CheckoutItem Component 1`] = `ShallowWrapper {}`;
